examples/question_answering_confidence.py

```diff
@@ -0,0 +1,123 @@
+from farm.evaluation.metrics import metrics_per_bin
+
+
+def question_answering_confidence():
+    ##########################
+    ########## Logging
+    ##########################
+    logger = logging.getLogger(__name__)
+    logging.basicConfig(
+        format="%(asctime)s - %(levelname)s - %(name)s -   %(message)s",
+        datefmt="%m/%d/%Y %H:%M:%S",
+        level=logging.INFO)
+    # reduce verbosity from transformers library
+    logging.getLogger('transformers').setLevel(logging.WARNING)
+
+    ##########################
+    ########## Settings
+    ##########################
+    set_all_seeds(seed=42)
+    device, n_gpu = initialize_device_settings(use_cuda=True)
+
+    lang_model = "deepset/roberta-base-squad2"
+    do_lower_case = False
+    batch_size = 80
+
+    data_dir = Path("../data/squad20")
+    # We use the same file for dev and test set only for demo purposes
+    dev_filename = "dev-v2.0.json"
+    test_filename = "dev-v2.0.json"
+    accuracy_at = 3 # accuracy at n is useful for answers inside long documents
+
+
+    # 1.Create a tokenizer
+    tokenizer = Tokenizer.load(
+        pretrained_model_name_or_path=lang_model,
+        do_lower_case=do_lower_case)
+
+    # 2. Create a DataProcessor that handles all the conversion from raw text into a pytorch Dataset
+    processor = SquadProcessor(
+        tokenizer=tokenizer,
+        max_seq_len=384,
+        label_list=["start_token", "end_token"],
+        metric="squad",
+        train_filename=None,
+        dev_filename=dev_filename,
+        test_filename=test_filename,
+        data_dir=data_dir,
+        doc_stride=192,
+    )
+
+    # 3. Create a DataSilo that loads several datasets (train/dev/test), provides DataLoaders for them and calculates a few descriptive statistics of our datasets
+    data_silo = DataSilo(
+        processor=processor,
+        batch_size=batch_size)
+
+
+    # 4. Load pre-trained question-answering model
+    model = AdaptiveModel.convert_from_transformers(lang_model, device=device, task_type="question_answering")
+    model.connect_heads_with_processor(data_silo.processor.tasks, require_labels=True)
+    # Number of predictions the model will make per Question.
+    # The multiple predictions are used for evaluating top n recall.
+    model.prediction_heads[0].n_best = accuracy_at
+
+    # 5. The calibration of model confidence scores sets one parameter, which is called temperature and can be accessed through the prediction_head
+    logger.info(f"Parameter used for temperature scaling of model confidence scores: {model.prediction_heads[0].temperature_for_confidence}")
+
+    # 6a. We can either manually set the temperature (default value is 1.0)...
+    model.prediction_heads[0].temperature_for_confidence = torch.nn.Parameter((torch.ones(1) * 1.0).to(device=device))
+
+    # 6b. ...or we can run the evaluator on the dev set and use it to calibrate confidence scores with a technique called temperature scaling.
```

Comment:
Can we add some doc how the updated temperature is used/inserted? Is it changing the param in place? How to insert this value (what is the value of it anyways? Is it printed) into a model loaded from huggingface?

Reply:
Okay, I add some comments. Regarding models from huggingface: In this example, the model is loaded via `AdaptiveModel.convert_from_transformers`, which works fine.